--- Config.pp	2014-03-07 17:06:14.658016331 -0600
+++ Config.pp	2014-03-07 17:08:11.619008334 -0600
@@ -168,7 +168,7 @@
 
 // By default, we specify the install/etc dir, which is where the
 // system-provided PRC files get copied to.
-#defer DEFAULT_PRC_DIR $[INSTALL_DIR]/etc
+#defer DEFAULT_PRC_DIR /etc/panda3d
 
 // You can specify the names of the environment variables that are
 // used to specify the search location(s) for prc files at runtime.
--- pptempl/Global.pp	2014-03-07 17:06:28.588015379 -0600
+++ pptempl/Global.pp	2014-03-07 17:07:11.362012454 -0600
@@ -930,8 +930,8 @@
 
 #define install_bin_dir $[or $[INSTALL_BIN_DIR],$[install_dir]/bin]
 #define install_data_dir $[or $[INSTALL_DATA_DIR],$[install_dir]/shared]
-#define install_igatedb_dir $[or $[INSTALL_IGATEDB_DIR],$[install_dir]/etc]
-#define install_config_dir $[or $[INSTALL_CONFIG_DIR],$[install_dir]/etc]
+#define install_igatedb_dir $[or $[INSTALL_IGATEDB_DIR],/etc/panda3d]
+#define install_config_dir $[or $[INSTALL_CONFIG_DIR],/etc/panda3d]
 #defer install_py_dir $[install_lib_dir]/$[PACKAGE]/$[DIRNAME]
 #defer install_py_package_dir $[install_lib_dir]/$[PACKAGE]
 
